### PR TITLE
feat(eslint-plugin-template): use-track-by-functions

### PR DESCRIPTION
**Pull request by Quramy**
## What I did

I implemented ESLint rule corresponding to http://codelyzer.com/rules/template-use-track-by-function/

### CODE REVIEW

File: packages/eslint-plugin-template/tests/rules/use-track-by-function.test.ts [hunk 1]
```diff
@@ -0,0 +1,147 @@
...
+          <li *ngFor="let item of [1, 2, 3];">
+              ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+            {{ item }}
+          </li>
+        </ul>
+      `,
+      messageId,
+    }),
+    //
+    // FIXME @angular/compiler can parse ngForTrackBy template text even if it's missing colon.
```
Comment (JamesHenry) [inline packages/eslint-plugin-template/tests/rules/use-track-by-function.test.ts:85]: Please could you remove this?

If there is work to be done in the angular compiler it should be tracked outside of the source code of this project

**Review by JamesHenry — CHANGES_REQUESTED** [review]
Just the one update, then we can merge this! Thanks a lot

**Comment by Quramy** [comment]
@JamesHenry  Thanks for reviewing. I've fixed pointed outs and pushed it.

**Comment by JamesHenry** [comment]
Released in 0.8.0-beta.5

**Comment by brandonpittman** [comment]
I'm on the `.7` beta and the rule isn't being picked up.

`Definition for rule '@angular-eslint/template-use-track-by-function' was not found`

Here's my config.

```js
module.exports = {
  env: {
    browser: true,
    node: true,
    es6: true,
  },
  extends: [
    "plugin:@angular-eslint/recommended",
    "plugin:@angular-eslint/template/process-inline-templates",
    "plugin:@typescript-eslint/recommended",
    "plugin:@typescript-eslint/recommended-requiring-type-checking",
    "plugin:jest-dom/recommended",
    "prettier",
    "prettier/@typescript-eslint",
  ],
  parser: "@typescript-eslint/parser",
  parserOptions: {
    project: "tsconfig.app.json",
    sourceType: "module",
  },
  plugins: [
    "@typescript-eslint",
    "@typescript-eslint/tslint",
    "@typescript-eslint/eslint-plugin",
    "eslint-plugin-tsdoc",
    "jest",
    "jest-dom",
    "unused-imports",
  ],
  rules: {
    "@angular-eslint/component-selector": "off",
    "@angular-eslint/directive-selector": "off",
    "@angular-eslint/no-empty-lifecycle-method": "off",
    "@angular-eslint/no-output-on-prefix": "off",
    "@angular-eslint/template-use-track-by-function": 2,
    "@typescript-eslint/explicit-module-boundary-types": "off",
    "@typescript-eslint/no-empty-function": "off",
    "@typescript-eslint/no-unsafe-assignment": "off",
    "@typescript-eslint/no-unsafe-call": "off",
    "@typescript-eslint/no-unsafe-member-access": "off",
    "@typescript-eslint/no-unsafe-return": "off",
    "@typescript-eslint/no-unused-vars": "off",
    "@typescript-eslint/unbound-method": ["error", { ignoreStatic: true }],
    "tsdoc/syntax": "error",
    "unused-imports/no-unused-imports-ts": "error",
    "unused-imports/no-unused-vars-ts": [
      "warn",
      {
        vars: "all",
        varsIgnorePattern: "^_",
        args: "after-used",
        argsIgnorePattern: "^_",
      },
    ],
  },
};
```

**Comment by brandonpittman** [comment]
@JamesHenry 

I'm on `0.8.0-beta..7` and the rule isn't being picked up.

`Definition for rule '@angular-eslint/template-use-track-by-function' was not found`

Here's my config.

```js
module.exports = {
  env: {
    browser: true,
    node: true,
    es6: true,
  },
  extends: [
    "plugin:@angular-eslint/recommended",
    "plugin:@angular-eslint/template/process-inline-templates",
    "plugin:@typescript-eslint/recommended",
    "plugin:@typescript-eslint/recommended-requiring-type-checking",
    "plugin:jest-dom/recommended",
    "prettier",
    "prettier/@typescript-eslint",
  ],
  parser: "@typescript-eslint/parser",
  parserOptions: {
    project: "tsconfig.app.json",
    sourceType: "module",
  },
  plugins: [
    "@typescript-eslint",
    "@typescript-eslint/tslint",
    "@typescript-eslint/eslint-plugin",
    "eslint-plugin-tsdoc",
    "jest",
    "jest-dom",
    "unused-imports",
  ],
  rules: {
    "@angular-eslint/component-selector": "off",
    "@angular-eslint/directive-selector": "off",
    "@angular-eslint/no-empty-lifecycle-method": "off",
    "@angular-eslint/no-output-on-prefix": "off",
    "@angular-eslint/template-use-track-by-function": 2,
    "@typescript-eslint/explicit-module-boundary-types": "off",
    "@typescript-eslint/no-empty-function": "off",
    "@typescript-eslint/no-unsafe-assignment": "off",
    "@typescript-eslint/no-unsafe-call": "off",
    "@typescript-eslint/no-unsafe-member-access": "off",
    "@typescript-eslint/no-unsafe-return": "off",
    "@typescript-eslint/no-unused-vars": "off",
    "@typescript-eslint/unbound-method": ["error", { ignoreStatic: true }],
    "tsdoc/syntax": "error",
    "unused-imports/no-unused-imports-ts": "error",
    "unused-imports/no-unused-vars-ts": [
      "warn",
      {
        vars: "all",
        varsIgnorePattern: "^_",
        args: "after-used",
        argsIgnorePattern: "^_",
      },
    ],
  },
};
```

Can anyone tell what I might be getting wrong?

**Pull request merged**